File: src/client/OrganizationBuilder.java
package client;

import exceptions.KeyboardInterruptException;
import lib.BufferedReaderWithQueueOfStreams;
import organization.Organization;
import organization.UserInteractiveOrganizationBuilder;

import java.io.IOException;
import java.time.LocalDate;

public class OrganizationBuilder {
    public static Organization constructOrganization(BufferedReaderWithQueueOfStreams reader, boolean prototypedFromAnother) throws KeyboardInterruptException, IOException {
        var organizationBuilder = new UserInteractiveOrganizationBuilder(reader, prototypedFromAnother);

        return new Organization(
                null,
                organizationBuilder.getName(),
                organizationBuilder.getCoordinates(),
                LocalDate.now(),
                organizationBuilder.getAnnualTurnover(),
                organizationBuilder.getFullName(),
                organizationBuilder.getEmployeesCount(),
                organizationBuilder.getOrganizationType(),
                organizationBuilder.getAddress()
        );
    }
}

File: src/application/OrganizationBuilder.kt
package application

import database.Address
import database.Organization
import lib.BufferedReaderWithQueueOfStreams
import java.time.LocalDate

object OrganizationBuilder {
    fun constructOrganization(reader: BufferedReaderWithQueueOfStreams, prototypedFromAnother: Boolean): Organization {
        val organizationBuilder = UserInteractiveOrganizationBuilder(reader, prototypedFromAnother)

        return Organization(
            null,
            organizationBuilder.name,
            organizationBuilder.coordinates,
            LocalDate.now(),
            organizationBuilder.annualTurnover,
            organizationBuilder.fullName,
            organizationBuilder.employeesCount,
            organizationBuilder.organizationType,
            organizationBuilder.address
        )
    }

    fun constructAddress(reader: BufferedReaderWithQueueOfStreams, prototypedFromAnother: Boolean): Address? {
        val organizationBuilder = UserInteractiveOrganizationBuilder(reader, prototypedFromAnother)
        return organizationBuilder.address
    }
}
